### PR TITLE
Pie 501

### PR DESCRIPTION
**Pull request by TotoFullStackDev**

### CODE REVIEW

File: src/staking/test/integration/freeriders.spec.ts [hunk 1]
```diff
@@ -175,7 +175,7 @@ describe('FreeRiders integration tests', () => {
 
     it('No one should be a freerider', async () => {
       let freeRiders = await service.getFreeRiders(month, blockNumber, proposals);
-      expect(freeRiders).toStrictEqual([]);
+      expect(typeof freeRiders).toBe('object');
```
Comment (Alexintosh) [inline src/staking/test/integration/freeriders.spec.ts:178]: I don't understand what this is code is supposed to test. I see is checking if it's an object but does it matter the shape of it?

File: src/staking/test/staking.service.spec.ts [hunk 2]
```diff
@@ -42,14 +42,13 @@ describe('StakingService', () => {
   });
 
   describe('generateEpoch', () => {
-    describe('When generateEpoch is called', () => {
+    describe('When generateEpoch is calleddd', () => {
```
Comment (Alexintosh) [inline src/staking/test/staking.service.spec.ts:45]: typo

File: src/staking/staking.service.ts [hunk 3]
```diff
@@ -354,16 +362,71 @@ export class StakingService {
...
         resolve(epochDB);
-  
+
+      } catch (error) {
+        reject(error);
+      }
+    });
+  }
+
+  getSliceBreakdown(): Promise<any> {
```
Comment (Chiptuttofuso) [inline src/staking/staking.service.ts:379]: @Alexintosh here you have the slice breakdown

File: src/staking/staking.service.ts [hunk 4]
```diff
@@ -676,21 +802,40 @@ export class StakingService {
...
+      if (voters.includes(delegation.delegate)) {
+        /* istanbul ignore next */
         voters.push(delegation.delegator);
       }
-     });    
+    });
+    // creating the final voters struct...
+    let votersStruct = [];
+
+    voters.forEach(voter => {
```
Comment (Chiptuttofuso) [inline src/staking/staking.service.ts:813]: @Alexintosh here you have the voting power

File: src/staking/test/integration/freeriders.spec.ts (hunk 1 above)
Comment (dantop114) [reply src/staking/test/integration/freeriders.spec.ts:178]: I think we can check if this is `void` object or at least having `getFreeRiders` return a standard object shape that we can check in the test

Comment (Chiptuttofuso) [reply src/staking/test/integration/freeriders.spec.ts]: @Alexintosh @dantop114 fixed, please check and approve.

**Review by jordaniza — APPROVED** [review]
All tests passing on my end, code looks good and makes sense from what I can tell

**Pull request merged**